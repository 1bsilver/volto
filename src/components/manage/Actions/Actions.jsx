/**
 * Actions component.
 * @module components/manage/Actions/Actions
 */

import React, { PropTypes, Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import { bindActionCreators } from 'redux';
import { filter, last, map } from 'lodash';
import { Dropdown, Icon } from 'semantic-ui-react';

import { cut, copy, copyContent, moveContent } from '../../../actions';
import config from '../../../config';

@connect(
  state => ({
    loaded: state.clipboard.request.loaded,
    action: state.clipboard.action,
    source: state.clipboard.source,
  }),
  dispatch => bindActionCreators({ cut, copy, copyContent, moveContent }, dispatch),
)
/**
 * Actions container class.
 * @class Actions
 * @extends Component
 */
export default class Actions extends Component {

  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    pathname: PropTypes.string.isRequired,
    loaded: PropTypes.bool.isRequired,
    action: PropTypes.string,
    source: PropTypes.string,
    cut: PropTypes.func.isRequired,
    copy: PropTypes.func.isRequired,
    copyContent: PropTypes.func.isRequired,
    moveContent: PropTypes.func.isRequired,
  }

  /**
   * Default properties
   * @property {Object} defaultProps Default properties.
   * @static
   */
  static defaultProps = {
    action: null,
    source: null,
  }

  /**
   * Constructor
   * @method constructor
   * @param {Object} props Component properties
   * @constructs Actions
   */
  constructor(props) {
    super(props);
    this.cut = this.cut.bind(this);
    this.copy = this.copy.bind(this);
    this.paste = this.paste.bind(this);
  }

  /**
   * Cut handler
   * @method cut
   * @returns {undefined}
   */
  cut() {
    this.props.cut(this.props.pathname);
  }

  /**
   * Copy handler
   * @method copy
   * @returns {undefined}
   */
  copy() {
    this.props.copy(this.props.pathname);
  }

  /**
   * Paste handler
   * @method paste
   * @returns {undefined}
   */
  paste() {
    if (this.props.action === 'copy') {
      this.props.copyContent(this.props.source, this.props.pathname);
    }
    if (this.props.action === 'cut') {
      this.props.moveContent(this.props.source, this.props.pathname);
    }
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    return (
      <Dropdown
        item
        trigger={<span><Icon name="lightning" /> Actions</span>}
        pointing="left"
      >
        <Dropdown.Menu>
          <Dropdown.Item
            icon="cut"
            text="Cut"
            onClick={this.cut}
          />
          <Dropdown.Item
            icon="copy"
            text="Copy"
            onClick={this.copy}
          />
          <Dropdown.Item
            icon="paste"
            text="Paste"
            onClick={this.paste}
          />
          <Link
            to={`${this.props.pathname}/delete`}
            className="item"
          >
            <Icon name="trash" />Delete
          </Link>
        </Dropdown.Menu>
      </Dropdown>
    );
  }
}
