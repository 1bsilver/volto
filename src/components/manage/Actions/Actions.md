Actions example source:

```jsx noeditor
const { Provider } = require('react-intl-redux');
const configureStore = require('redux-mock-store').default;
import BrowserRouter from 'react-router-dom/BrowserRouter';
import Actions from './Actions';
import StaticRouter from 'react-router-dom/StaticRouter';
const store = configureStore()({
  userSession: {
    login: {},
  },
  intl: {
    locale: 'en',
    messages: {},
  },
  users: {
    reset: {
      loading: '',
    },
  },
  actions: {},
  content: { get: {} },
});

<div className={'rsg--pre-42'}>
  <Provider store={store}>
    <StaticRouter>
      <Actions
        listActions={() => {}}
        pathname=""
        location={{ search: '/hello', pathname: '/' }}
        getContent={() => {}}
        versionId=""
        content={{
          layout: 'PLone',
          allow_discussion: true,
          title: 'plone',
          description: 'plone CMS',
          '@type': '@Document',
          is_folderish: true,
          subjects: [],
        }}
      />
    </StaticRouter>
  </Provider>
</div>;
```
