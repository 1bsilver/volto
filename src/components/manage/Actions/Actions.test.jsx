import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import Actions from './Actions';

const mockStore = configureStore();

describe('Actions', () => {
  it('renders an actions component', () => {
    const store = mockStore({ clipboard: {} });
    const component = renderer.create(
      <Provider store={store}>
        <Actions pathname="/test" />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
