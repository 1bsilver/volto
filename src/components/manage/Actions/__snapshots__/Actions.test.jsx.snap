// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions renders an actions component 1`] = `
<div
  aria-busy={undefined}
  aria-disabled={undefined}
  aria-expanded={false}
  aria-multiselectable={undefined}
  className="ui item left pointing dropdown"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  role="listbox"
  tabIndex={0}
>
  <span>
    <i
      aria-hidden="true"
      className="lightning icon"
    />
     Actions
  </span>
  <i
    aria-hidden="true"
    className="dropdown icon"
  />
  <div
    className="menu transition"
  >
    <div
      aria-checked={undefined}
      aria-disabled={undefined}
      aria-selected={undefined}
      className="item"
      onClick={[Function]}
      role="option"
    >
      <i
        aria-hidden="true"
        className="cut icon"
      />
      <span
        className="text"
      >
        Cut
      </span>
    </div>
    <div
      aria-checked={undefined}
      aria-disabled={undefined}
      aria-selected={undefined}
      className="item"
      onClick={[Function]}
      role="option"
    >
      <i
        aria-hidden="true"
        className="copy icon"
      />
      <span
        className="text"
      >
        Copy
      </span>
    </div>
    <div
      aria-checked={undefined}
      aria-disabled={true}
      aria-selected={undefined}
      className="disabled item"
      onClick={[Function]}
      role="option"
    >
      <i
        aria-hidden="true"
        className="paste icon"
      />
      <span
        className="text"
      >
        Paste
      </span>
    </div>
    <a
      className="item"
      onClick={[Function]}
      style={Object {}}
    >
      <i
        aria-hidden="true"
        className="trash icon"
      />
      Delete
    </a>
  </div>
</div>
`;
